Formula/firefoxpwa.rb

```diff
@@ -0,0 +1,77 @@
+class Firefoxpwa < Formula
+  desc "Tool to install, manage and use Progressive Web Apps in Mozilla Firefox"
+  homepage "https://github.com/filips123/FirefoxPWA"
+  url "https://github.com/filips123/FirefoxPWA/archive/refs/tags/v0.5.0.tar.gz"
+  sha256 "40004c7a268436173813024672ea23599e387766973d36d0ee0b81793a20b169"
+  license "MPL-2.0"
+  head "https://github.com/filips123/FirefoxPWA.git"
+
+  depends_on "rust" => :build
+
+  def install
+    cd "native"
+
+    # Set the correct version in the source files
+    inreplace "Cargo.toml",
+              "version = \"0.0.0\"",
+              "version = \"#{version}\""
+    inreplace "userchrome/profile/chrome/pwa/chrome.jsm",
+              "DISTRIBUTION_VERSION = '0.0.0'",
+              "DISTRIBUTION_VERSION = '#{version}'"
+
+    # Set the path in the manifest to the Homebrew libexec directory
+    # This is needed so the program respects Homebrew's installation hierarchy
+    # We can use macOS manifest for Linux because the only difference is a default path
+    inreplace "manifests/macos.json",
+              "/usr/local/libexec/firefoxpwa-connector",
+              "#{libexec}/firefoxpwa-connector"
+
+    # Set the system data path to the Homebrew share directory
+    # This is needed so the program respects Homebrew's installation hierarchy
+    ENV["FFPWA_SYSDATA"] = share
+
+    # Build the native part
+    # We cannot use "cargo install" to automatically install it
+    # Because the package contains multiple binary and other files
+    system "cargo", "build", "--release"
```

Comment:
We should also pass `--locked` to `cargo`.

I'm not sure I understand the comment here. What happens if we do `cargo install`?

Reply:
Project needs to install two binaries:

* `firefoxpwa` - Normal CLI program, can be executed by user, so should be installed into `bin` directory.
* `firefoxpwa-connector` - Additional program that communicates with Firefox extension, it is executed by Firefox using [native messaging](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_messaging) protocol, should not be executed by user directly, so it should be installed into `libexec` directory.

I used `cargo build` to build both binaries and then manually copy them to correct locations. However, if I use `cargo install`, both binaries will be installed into `bin` directory.

Although project will still work when both binaries are in `bin`, I think that because `firefoxpwa-connector` is internal program that should not be executed by users, it should be outside `PATH` and in `libexec`.

Reply:
Let's just do `system "cargo", "install", *std_cargo_args` then, and then do
```ruby
libexec.install bin/"firefoxpwa-connector"
```
to move `firefoxpwa-connector` to the right place.